gatt/gatt.py

```diff
@@ -96,15 +93,21 @@ def devices(self):
     def start_discovery(self, service_uuids=[]):
         """Starts a discovery for BLE devices with given service UUIDs.
 
-        Args:
-            service_uuids: Filters the search to only return devices with given UUIDs.
+        :param service_uuids: Filters the search to only return devices with given UUIDs.
         """
 
         discovery_filter = {'Transport': 'le'}
         if service_uuids:  # D-Bus doesn't like empty lists, it needs to guess the type
             discovery_filter['UUIDs'] = service_uuids
-        self.adapter.SetDiscoveryFilter(discovery_filter)
-        self.adapter.StartDiscovery()
+
+        try:
+            self.adapter.SetDiscoveryFilter(discovery_filter)
+            self.adapter.StartDiscovery()
+        except dbus.exceptions.DBusException as e:
+            if e.get_dbus_name() == 'org.bluez.Error.NotReady':
+                raise errors.NotReady("Bluetooth adapter not ready. Run 'hciconfig " + self.adapter_name + " up'.")
```

Comment:
Why you don't want to use one of the many string interpolation methods that Python has instead of `+`? :)